circe-typify/src/main/scala/CirceParsed.scala

```diff
@@ -0,0 +1,59 @@
+package circe.api.libs.json.typify
+
+import io.circe.Json.{JArray, JBoolean, JNull, JNumber, JObject, JString}
+import io.circe.syntax._
+import io.circe.{Decoder, DecodingFailure, HCursor, Json}
+import scala.reflect.ClassTag
+import scalaz.std.list._
+import scalaz.std.option._
+import scalaz.std.string._
+import scalaz.syntax.id._
+import scalaz.syntax.nel._
+import scalaz.syntax.std.option._
+import scalaz.syntax.std.string._
+import scalaz.syntax.traverse._
+import scalaz.syntax.validation._
+import scalaz.{\/, NonEmptyList, ValidationNel}
+import typify.{CanParse, Parsed, ParseError}
+
+private[this] object helper {
+  def tpv[T](ct: String)(e: Either[DecodingFailure, T], kO: Option[String]=None): ValidationNel[ParseError, T] = e match {
+    case Left(err) => ParseError(kO.getOrElse("_root_"), s"Could not be ${kO.map(_ => "parsed").getOrElse("interpreted")} as $ct:\n\t$err").failureNel[T]
+    case Right(t) => t.successNel[ParseError]
+  }
+  def mk[T: ClassTag](ct: String)(implicit D: Decoder[T]) = new CanParse[T, Json] {
```

Comment:
I don't think you need the `ClassTag` here

Reply:
you are correct - updated